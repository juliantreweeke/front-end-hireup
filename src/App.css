.App {
  text-align: center;
}

.App-logo {
  float:left;
  margin-left:15%;
  height: 45px;
}

.App-header {
  background-color:#EF1B1F;
  height: 49px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

#search-bar {
  float:right;
  margin-top:7px;
  margin-right:17%;
  width:240px;
  height:35px;
  border-radius:5px;
  border:none;
}

#search-bar:focus {
  border:blue 0.5px solid;
}

/* Chrome/Opera/Safari */
#search-bar::-webkit-input-placeholder {
  color: rgba(142, 149, 168, 0.9);
  padding-left:20px;
}

/* Firefox 19+ */
#search-bar::-moz-placeholder {
  color: rgba(142, 149, 168, 0.9);
  padding-left:20px;
}

.loading {
    background: url('./img/spinner-small.svg') no-repeat right center;
    /* background: url(http://www.xiconeditor.com/image/icons/loading.gif) no-repeat right center; */
    background-color: white;
}
/* mobile burger menu styles */

.mobile-menu {
  display:none;
  position: relative;
  top:-5px;
  right:5px;
  cursor: pointer;
}

.mobile-menu span {
    border-radius:3px;
    content: "";
    position: absolute;
    left: 30px;
    width: 32px;
    border: 3px solid #fff;
}

.mobile-menu span:nth-child(1) {
  top: 18px;
}

.mobile-menu span:nth-child(2) {
  top: 27px;
}

.mobile-menu span:nth-child(3) {
  top: 36px;
}












/* Smartphones (landscape) ----------- */
@media only screen and (max-width : 768px) {

  /* on mobile view - show burger menu option */
  .mobile-menu {
    display:block;
  }

  .App-logo {
   position: absolute;
   top: 5%;
   width: 200px;
   height: 78px;
   margin-top: -33px;
   margin-left: -100px;
  }

  #search-bar {
    font-size:20px;
    position: absolute;
    border: 1px solid rgba(142, 149, 168, 0.3);
    margin-top: 100px;
    margin-left: -180px;
    width:360px;
    height:60px;
    background-color:#F9F9F9;
  }

  .loading  {
    background: url('./img/spinner.svg') no-repeat right center;
  }






  #search-bar::-webkit-input-placeholder {
   font-size:20px;
   padding-left:30px;
  }

  /* Firefox 19+ */
  #search-bar::-moz-placeholder {
    font-size:20px;
  }

}
