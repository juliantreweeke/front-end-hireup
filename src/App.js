import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';


const url = 'http://localhost:1111/characters';
let search = 'iron';
const url_search = `http://localhost:1111/characters?nameStartsWith=${search}`;






    



class App extends Component {
  
  constructor(props) {
    super(props);
    this.state = {search: 'iron', success:'false'
    };
    
    this.auto_suggest = this.auto_suggest.bind(this);
  
  }
  
  
  
  search(query){
    
    fetch(query)
    .then(results => {
      return results.json();
    }).then(data => {
        console.log(data.results.length);
        
        data.results.length > 0 ? data.results.map((character,i) => {
          
          console.log(character.name);
        }) : console.log("whoops can't find the name!");    
        
    })
      
    
  }
  
 auto_suggest(e){
   this.setState({search: e.target.value});
   this.search(this.state.search);
   
   this.state.search === "iron" ? ()=>{ this.setState({success:true}) }:null;
 }
  
  
  


  
  
  
  
  
  
  
  
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
          <input onChange={(e)=>{ this.auto_suggest(e) } } id="searchbar" type="text" placeholder="your search"></input>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <h1>{this.state.search}</h1>
        {this.state.success}
        
        {this.state.success
          ? <div>found</div>
          : <div>not found</div>}
        
        
        
        
        
      </div>
    );
  }
}

export default App;
