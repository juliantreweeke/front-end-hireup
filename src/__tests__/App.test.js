import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
Enzyme.configure({ adapter: new Adapter() });
import { mount } from 'enzyme';
import { shallow } from 'enzyme';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

describe('<App />', () => {

  it('renders 1 <App /> component', () => {
    const component = shallow(<App name="app"/>);
    expect(component).toHaveLength(1);
  });

  describe('it renders props correctly', () => {
    const component = shallow(<App name="app" />);
    expect(component.instance().props.name).toBe('app');
  })


  it('renders an `.logo`', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper.find('.App-logo')).toHaveLength(1);
  });

  it('renders an `.App-header`', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper.find('.App-header')).toHaveLength(1);
  });

  it('renders an `#search-bar`', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper.find('#search-bar')).toHaveLength(1);
  });

  it('renders an `.mobile-menu`', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper.find('.mobile-menu')).toHaveLength(1);
  });

  it('renders and matches our snapshot', () => {
  const component = renderer.create(
    <App name="app" />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});





});
