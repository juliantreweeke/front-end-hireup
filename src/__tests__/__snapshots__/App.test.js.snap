// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders and matches our snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.png"
    />
    <div
      className="mobile-menu"
    >
      <span />
      <span />
      <span />
    </div>
    <input
      className=""
      id="search-bar"
      onChange={[Function]}
      placeholder="Search.."
      type="text"
    />
  </header>
</div>
`;
